client/pages/_app.jsx

```diff
@@ -16,4 +19,32 @@ class MyApp extends App {
   }
 }
 
-export default withApolloClient(MyApp);
+const withStore = App => props => (
+  <StoreProvider>
+    <App {...props} />
+  </StoreProvider>
+);
+const PUBLIC_PAGES = ["/", "/login", "/signup"];
+const withAuthorization = App => {
```

Comment:
This is a HOC which authorizes users depending on which page is requested.
*This logic is still NOT working and be implemented properly*

Reply:
Will the login and signup pages still display normally and be functional?

Reply:
Yes, but if you are not happy with this, I can port this part to a new PR.

Reply:
I think as long as the pages are functional for now, it's fine.